### PR TITLE
Add Prometheus Pushgateway as an optional service

### PR DESCRIPTION
**Pull request by suiguoxin**
- Add an optional service Prometheus Pushagteway
- add a container `metrics-cleaner` to clean Pushgateway metrics by fixed interval
- add prometheus-pushgateway in job-exporter
- set `honor_lables` as true in Prometheus

### CODE REVIEW

File: src/prometheus-pushgateway/deploy/prometheus-pushgateway.yaml.template [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        image: {{ cluster_cfg["cluster"]["docker-registry"]["prefix"] }}prometheus-pushgateway:{{ cluster_cfg["cluster"]["docker-registry"]["tag"] }}
+        imagePullPolicy: Always
+        {%- if cluster_cfg['cluster']['common']['qos-switch'] == "true" %}
+        resources:
+          limits:
+            memory: "2Gi"
+            cpu: "3000m"
+          requests:
+            memory: "512Mi"
+            cpu: "2000m"
```
Comment (Binyang2014) [inline src/prometheus-pushgateway/deploy/prometheus-pushgateway.yaml.template:42]: Can we change it to 1 or 0.5?

**Review by Binyang2014 — COMMENTED** [review]
Please change job-exporter, then we can monitor the push-gateway cpu/memory usage

**Comment by suiguoxin** [comment]
> Please change job-exporter, then we can monitor the push-gateway cpu/memory usage

It's added [here](https://github.com/microsoft/pai/pull/5590/files#diff-e171e1883846a5fc98f723b7fcbeead63a70c2641a7cff45b25cf04b05ac0c0dR542-R548)

File: src/prometheus-pushgateway/deploy/prometheus-pushgateway.yaml.template (hunk 1 above)
Comment (suiguoxin) [reply src/prometheus-pushgateway/deploy/prometheus-pushgateway.yaml.template]: > Can we change it to 1 or 0.5?

Updated

**Review by Binyang2014 — APPROVED** [review]

**Pull request merged**